### PR TITLE
Settingspanel

### LINKED ISSUE #52: Preferences/settings UI for telemetry, crash reporting, privacy policy
Preferences/settings UI for telemetry, crash reporting, privacy policy

### PR DESCRIPTION
**Pull request by keianhzo**
Resolves #52

### CODE REVIEW

**Comment by bluemarvin** [comment]
Downloaded branch and ran locally. Looks good. Just quickly glanced over source and didn't see anything glaring. I'll leave the in depth code review to @MortimerGoro :)

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 1]
```diff
@@ -383,6 +397,102 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
...
+    //Settings
+    private void showSettingsWidget() {
+        if (mSettingsWidget == null) {
+            mSettingsWidget = new SettingsWidget(this);
+            mSettingsWidget.getPlacement().parentHandle = mBrowserWidget.getHandle();
+            mSettingsWidget.setDelegate(new SettingsWidget.Delegate() {
+                @Override
+                public void onSettingsCrashReportingChange(boolean isEnabled) {
+                    SettingsStore.getInstance(VRBrowserActivity.this).setCrashReportingEnabled(isEnabled);
+                    SessionStore.get().getSettings().setRemoteDebuggingEnabled(isEnabled)
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Should we enable remote debugging when crash reporting is enabled?

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 2]
```diff
@@ -383,6 +397,102 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         }
     }
 
+    //Settings
+    private void showSettingsWidget() {
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: I'd prefer that we move all this code to the SettingsWidget class itself. We don't want VRBrowerActivity class become very big. You can access the WidgetManager from any widget to show/hide things.

(e.g. Ideally browser window or navigation bar just calls mSettingsWidget.show())

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 3]
```diff
@@ -383,6 +397,102 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
...
+            });
+            addWidget(mSettingsWidget);
+        }
+        else {
+            mSettingsWidget.getPlacement().visible = true;
+            updateWidget(mSettingsWidget);
+        }
+    }
+
+    private void hideSettingsWidget() {
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Move this too

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 4]
```diff
@@ -383,6 +397,102 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
...
+        if (mSettingsWidget != null && mSettingsWidget.getVisibility() == View.VISIBLE) {
+            mSettingsWidget.getPlacement().visible = false;
+            updateWidget(mSettingsWidget);
+        }
+    }
+
+    private void showCrashReportingWidget() {
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: move this too

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 5]
```diff
@@ -383,6 +397,102 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
...
+    }
+
+    private void hideCrashReportingWidget() {
+        if (mCrashReportingWidget != null && mCrashReportingWidget.getVisibility() == View.VISIBLE) {
+            mCrashReportingWidget.getPlacement().visible = false;
+            updateWidget(mCrashReportingWidget);
+        }
+    }
+
+    private void handleRestartApp() {
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: move this too

File: app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java [hunk 6]
```diff
@@ -88,6 +89,13 @@ public void setContext(Context aContext) {
         }
     }
 
+    public GeckoRuntimeSettings getSettings() {
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java]: Rename this to getRuntimeSettings? We use getSettings() in other places for GeckoSession.Settings

File: app/src/common/shared/org/mozilla/vrbrowser/ui/SettingsWidget.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            versionText.setText(String.format(getResources().getString(R.string.settings_version), pInfo.versionName));
+
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    void initializeWidgetPlacement(WidgetPlacement aPlacement) {
+        aPlacement.width = 435;
```
Comment (MortimerGoro) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/SettingsWidget.java]: add width & height to dimens.xml

**Comment by MortimerGoro** [comment]
Looks good! Almost ready to land.

Apart from the requested changes please squash the commits.

Thanks!

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java (hunk 1 above)
Comment (bluemarvin) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Remote debugging will need to be a separate setting and is not related to crash reporting.

**Comment by bluemarvin** [comment]
The person doing the merge can "Squash and merge" which is the method I prefer for merging PRs since it doesn't add an additional merge point to the history.

**Comment by keianhzo** [comment]
Ready to be merged

**Pull request merged**